test/integration/containerized/conftest.py

```diff
@@ -60,7 +60,7 @@ def run(args, *a, **kw):
         })
 
         try:
-            ret = CompletedProcessProxy(subprocess.run(args, shell=True, *a, **kw))
+            ret = CompletedProcessProxy(subprocess.run(' '.join(args), shell=True, *a, **kw))
```

Comment:
Some people don't like to see `shell=True`. I do not expression a position on that. But if you are going to use shell...

https://docs.python.org/3/library/subprocess.html

```python
>>> subprocess.run(["ls", "-l"])  # doesn't capture output
CompletedProcess(args=['ls', '-l'], returncode=0)

>>> subprocess.run("exit 1", shell=True, check=True)
Traceback (most recent call last):
  ...
subprocess.CalledProcessError: Command 'exit 1' returned non-zero exit status 1
```

It takes a list if you don't pass `shell=True`. If you do use shell mode, docs say to pass a string, but if you pass a list anyway, then it runs only the first element of the list. In our case `ansible-runner` by itself exits with code 0, so the fact that this was happening appears to not be known by anyone.

Reply:
woops